### ISSUE TITLE
the training time

### ISSUE THREAD

**Issue by KN-Zhang** [Nanne/pytorch-NetVlad#72]
Hi~ Did you initialize the centroids at the beginning of training? I found ``net_vlad.init_params(clsts, traindescs) `` in ``main.py`` would not be performed because ``if not opt.resume:'' cannot be satisfied.
Also, could you tell me how much dose it take to train the model? I have trained 36 epoches for 30 hours (without centroid initialization) and the loss function seems to have converged, but I got poor results of:
Recall@1: 0.4201
Recall@5: 0.6317
Recall@10: 0.7212
Recall@20: 0.8026.

I don't know what the problem is...

**Comment by Nanne** [comment]
As described in the README: `In order to initialise the NetVlad layer it is necessary to first run main.py with the correct settings and --mode=cluster.`

Not initialising the centroids is probably why you get poor performance, this is a necessary step. As for the training speed, please see this previous discussion: https://github.com/Nanne/pytorch-NetVlad/issues/2
